corehq/apps/users/tests/test_user_invitation.py

```diff
@@ -0,0 +1,96 @@
+import datetime
+
+from unittest.mock import Mock
+
+from django.test import TestCase
+
+from corehq.apps.users.models import Invitation
+from corehq.apps.users.views.web import UserInvitationView, WebUserInvitationForm
+
+from django import forms
+
+
+class StubbedWebUserInvitationForm(WebUserInvitationForm):
+
+    def __init__(self, *args, **kwargs):
+        self.request_email = kwargs.pop('request_email', False)
+        super().__init__(*args, **kwargs)
+
+    @property
+    def cleaned_data(self):
+        return {"email": self.request_email}
+
+
+class TestUserInvitation(TestCase):
+
+    def test_redirect_if_invite_does_not_exist(self):
+        request = Mock()
+        non_existing_uuid = "e1bd37f5-9ff8-4853-b953-fd75483a0ec7"
+        domain = "domain"
+
+        response = UserInvitationView()(request, non_existing_uuid, domain=domain)
+        self.assertEqual(302, response.status_code)
+        self.assertEqual("/accounts/login/", response.url)
+
+    def test_redirect_if_invite_is_already_accepted(self):
+        request = Mock()
+        invite_uuid = "e1bd37f5-9ff8-4853-b953-fd75483a0ec7"
+        domain = "domain"
+
+        invite = Invitation.objects.create(
+            uuid=invite_uuid,
+            domain=domain,
+            is_accepted=True,
+            invited_on=datetime.date(2023, 9, 1)
+        )
+        invite.save()
+
+        response = UserInvitationView()(request, invite_uuid, domain=domain)
+        self.assertEqual(302, response.status_code)
+        self.assertEqual("/accounts/login/", response.url)
+
+    def test_redirect_if_invite_email_does_not_match(self):
+        form = StubbedWebUserInvitationForm(
+            {
+                "email": "other_test@dimagi.com",
+                "full_name": "asdf",
+                "password": "pass",
+            },
+            is_sso=False,
+            allow_invite_email_only=True,
+            invite_email="test@dimagi.com",
+            request_email="other_test@dimagi.com",)
```

Comment:
nit: typical wrapping in this style would put the closing paren on a new line and dedented to the level of the line with the opening paren.
```suggestion
            request_email="other_test@dimagi.com",
        )
```
There are a few other places in this PR where this comment also applies.